_episodes/10-zonal-statistics.md

```diff
@@ -0,0 +1,188 @@
+
+We convert the output of the `rasterio.features.rasterize` function, which generates a numpy array `np.ndarray`, to `xarray.DataArray` which will be used further:
+~~~
+import xarray as xr
+field_cropped_raster_xarr = xr.DataArray(field_cropped_raster)
+~~~
+{: .language-python}
+
+# Calculate zonal statistics 
+
+In order to calculate the statistics for each crop zone, we call the function, `xrspatial.zonal_stats`. The `xrspatial.zonal_stats` function takes as input `zones`, a 2D `xarray.DataArray`, that defines different zones, and `values`, a 2D `xarray.DataArray` providing input values for calculating statistics. We use the `squeeze()` function in order to reduce our raster data `ndvi` dimension to 2D by removing the singular `band` dimension:
+
+~~~
+ndvi_sq = ndvi.squeeze()
+~~~
+{: .language-python}
+
+Then we call the `zonal_stats` function with `cropfield_raster_xarr` as our classifier and the 2D raster with our values of interest `ndvi_sq` to obtain the NDVI statistics for each crop type:
+
+~~~
+from xrspatial import zonal_stats
+zonal_stats(cropfield_raster_xarr, ndvi_sq)
+~~~
+{: .language-python}
+
+~~~
+	zone	mean	max	min	sum	std	var	count
+0	0	0.266531	0.999579	-0.998648	38887.648438	0.409970	0.168075	145903.0
+1	259	0.520282	0.885242	0.289196	449.003052	0.111205	0.012366	863.0
+2	265	0.775609	0.925955	0.060755	66478.976562	0.091089	0.008297	85712.0
+3	266	0.794128	0.918048	0.544686	1037.925781	0.074009	0.005477	1307.0
+4	331	0.703056	0.905304	0.142226	10725.819336	0.102255	0.010456	15256.0
+5	332	0.681699	0.849158	0.178113	321.080261	0.123633	0.015285	471.0
+6	335	0.648063	0.865804	0.239661	313.662598	0.146582	0.021486	484.0
+7	863	0.388575	0.510572	0.185987	1.165724	0.144245	0.020807	3.0
+~~~
+{: .output}
+
+The `zonal_stats` function calculates the minimum, maximum, and sum for each zone along with statistical measures such as the mean, variance and standard deviation for each rasterized vector zone. In our raster data-set `zone = 0`, corresponding to non-crop areas, has the highest count followed by `zone = 265` which corresponds to 'Grasland, blijvend' or 'Grassland, permanent'. The highest mean NDVI is observed for `zone = 266` for 'Grasslands, temporary' with the lowest mean, aside from non-crop area, going to `zone = 863` representing 'Forest without replanting obligation'. Thus, the `zonal_stats` function can be used to analyse and understand different sections of our raster data. The definition of the zones can be derived from vector data or from classified raster data as presented in the challenge below: 
+
+> ## Challenge: Calculate zonal statistics for zones defined by `ndvi_classified`
+> 
+> Let's calculate NDVI zonal statistics for the different zones as classified by `ndvi_classified` in the previous episode .
```

Comment:
Same as above, I think it would be better to not assume  `ndvi_classified` to be already loaded:
```suggestion
> Let's calculate NDVI zonal statistics for the different zones as classified and saved to `NDVI_classified.tif` in the previous episode.
```
(also need to add `ndvi_classified = rioxarray.open_rasterio("NDVI_classified.tif")` below)

Reply:
Resolved as before.